web/controllers/v1/user_controller.ex

```diff
@@ -56,4 +58,34 @@ defmodule Cog.V1.UserController do
     Repo.delete!(user)
     send_resp(conn, :no_content, "")
   end
+
+  ###############################################
+  # Plug function - local only to user controller
+  ###############################################
+  @spec check_self_updating(Plug.Conn.t, Plug.Conn.t) :: Plug.Conn.t
+  def check_self_updating(conn, opts \\ []) do
+    if determine_self_updating(conn) do
+      conn
+    else
+      plug_opts = Cog.Plug.Authorization.init(opts)
+      case Cog.Plug.Authorization.call(conn, plug_opts) do
+        {:error, error} ->
+          error
```

Comment:
Is this matching a specific error? It looks like in the authorization plug we always return a conn, which has been halted if the user is not authorized.